error.go

```diff
@@ -36,6 +46,52 @@ func (e internalError) InternalError() error {
 	return e.error
 }
 
+// SetCode set the error's translation code.
+func (e Error) SetCode(code string) Error {
+	e.code = code
+
+	return e
+}
+
+// Code get the error's translation code.
+func (e Error) Code() string {
+	return e.code
+}
+
+// SetParams set the error's params.
+func (e Error) SetParams(params map[string]interface{}) Error {
```

Comment:
Can we also add `AddParam(name string, value interface{})`? This will make setting a single parameter (hopefully a main use scenario) easier.

Reply:
Yea, Sure